Bug fix for clicksend

JaxomCS opened the pull request:
Current version causes 500 error since it is sending an array of from numbers to ClickSend. Changing the from number to 'hass' identifies all messages as coming from Home Assistant making them more recognisable and removes the bug.

## Description:


**Related issue (if applicable):** fixes #<home-assistant issue number goes here>

**Pull request in [home-assistant.io](https://github.com/home-assistant/home-assistant.io) with documentation (if applicable):** home-assistant/home-assistant.io#<home-assistant.io PR number goes here>

## Checklist:
  - [✓] The code change is tested and works locally.
  - [✓] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
  - [✓] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [✓] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [✓] New or updated dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [✓] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [✓] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

JaxomCS committed:
Bug fix

Current version causes 500 error since it is sending an array of from numbers to ClickSend. Changing the from number to 'hass' identifies all messages as coming from Home Assistant making them more recognisable and removes the bug.

Danielhiversen commented:
You also have to remove the `sender` from the config and update the documentation.

JaxomCS committed:
Amendment

Changed it to use 'hass' as the default instead of defaulting to the recipient which is the array. Would have worked if users set their own name but users who were using the default were experiencing the issue.

Danielhiversen commented on homeassistant/components/notify/clicksend.py:
```diff
@@ -29,7 +29,7 @@ def validate_sender(config):
     if CONF_SENDER in config:
         return config
-    config[CONF_SENDER] = config[CONF_RECIPIENT]
+    config[CONF_SENDER] = 'hass'
```
'hass' should be extracted as a constant

JaxomCS replied:
Would you want this added to the homeassistant.const as I cannot really see any place for it in there and that seems to be the only extracted constants file I can see.

Danielhiversen replied:
No, just add it in the top of the file

Danielhiversen reviewed (changes requested):
Thanks, just a minor comment

JaxomCS committed:
Added DEFAULT_SENDER variable

balloob commented:
Is it really a breaking change if the current integration does not work?

Tagged it for 81 so it can go out on Friday.
